src/coreComponents/codingUtilities/TableLayout.hpp

```diff
@@ -0,0 +1,148 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file TableLayout.hpp
+ */
+
+#ifndef GEOS_COMMON_TABLELAYOUT_HPP
+#define GEOS_COMMON_TABLELAYOUT_HPP
+
+#include "common/DataTypes.hpp"
+
+namespace geos
+{
+
+class TableLayout
+{
+
+public:
+  enum Alignment { right, left, center };
+
+  enum MarginValue : integer
+  {
+    tiny = 0,
+    small = 1,
+    medium = 2,
+    large = 3
+  };
+
+  /**
+   * @brief Enumeration for table sections.
+   */
+  enum Section { header, values };
+
+  /**
+   * @brief Structure to set up each colum parameters.
+   */
+  struct ColumnParam
+  {
+
+    string columnName;
+    // Alignment for a column. By default aligned to the right side
+    Alignment alignment = Alignment::right;
+    // A boolean to display a colummn
+    bool enabled = true;
+    // Vector containing substring column name delimited by "\n"
+    std::vector< string > splitColumnName;
+
+    /**
+     * @brief Construct a ColumnParam object with the specified name and alignment.
+     * @param name The name of the column
+     * @param align The alignment of the column
+     */
+    ColumnParam( std::string const & name, Alignment align )
+      : columnName( name ), alignment( align )
+    {}
+
+    /**
+     * @brief Construct a ColumnParam object with the specified name, alignment, and display flag.
+     * @param name The name of the column
+     * @param align The alignment of the column
+     * @param display Flag indicating whether the column is enabled
+     */
+    ColumnParam( std::string const & name, Alignment align, bool display )
+      : columnName( name ), alignment( align ), enabled( display )
+    {}
+  };
+
+  /**
+   * @brief Struct for a column.
+   */
+  struct Column
+  {
+    ColumnParam parameter;
+    // A vector containing all column values
+    std::vector< string > columnValues;
+    // The largest string in the column
+    string m_maxStringSize;
+  };
+
+  TableLayout() = default;
```

Comment:
Do you need this? Now that you can't modify it after the construction (I'm OK with that), do you need a dummy constructor? If not, maybe `delete` is instead?

Reply:
Indeed I don't need this, I initialized the CSV constructors incorrectly, fixed